lib/freedom_patches/net_http.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+module NetHTTPPatch
+  OPEN_TIMEOUT = 5
+  READ_TIMEOUT = 10
+  WRITE_TIMEOUT = 5
+
+  # By default Net::HTTP will retry 1 time on idempotent requests but we can't afford that while processing a request
+  # so setting it to 0
+  MAX_RETIRES = 0
+
+  def initialize(*args, &block)
+    super(*args, &block)
+
+    ## START PATCH
+    if Thread.current[Middleware::ProcessingRequest::PROCESSING_REQUEST_THREAD_KEY]
```

Comment:
What is not very clear to me is why we want to limit the patch to when processing the request. I assume this is a way to target just Unicorn and not Sidekiq? In this case, why cannot we patch `Net::HTTP` instances from Unicorn only? 🤔

Reply:
A unicorn process can be running background threads too and those do not require the same strict limit as when we are processing a request. The main objective here is to reduce the risk of an external request causing a Unicorn worker to timeout.